### PR TITLE
Svc 2240 start to support unity builds on top of bitrise

### PR DESCRIPTION
**Pull request by moczolaszlo**

### CODE REVIEW

**Review by roland-spekker-bitrise — APPROVED** [review]

File: source/javascripts/controllers/_LicensesController.js.erb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+					var pools = response.licensePools;
+					if (pools.length > 0) {
+						_.each(pools, function(pool) {
+							$scope.licenses.push(pool);
+						});
+					}
+					else {
+						$scope.licenses = [];
+					}
+					$scope.$apply()
```
Comment (erosdome) [inline source/javascripts/controllers/_LicensesController.js.erb:35]: Is this needed? Afaik AngularJS runs a digest loop properly when a service's async request is finished.

File: source/javascripts/services/_appService.js.erb [hunk 2]
```diff
@@ -283,6 +287,27 @@
 					appService.savedDefaultMachineType = angular.copy(appService.defaultMachineType);
 				}
 
+				if (
+					requestService.mode === "website" &&
+					appService.defaultLicensePoolId
```
Comment (erosdome) [inline source/javascripts/services/_appService.js.erb:292]: If the user unsets the default license pool ID, this `if` will not run, thus [unsetting on save](https://github.com/bitrise-io/bitrise-workflow-editor/pull/802/files#diff-34c45354114ff878b352674d21c5a0c6914d86abbf5b64974b99186219b31550R305) will not happen.

My first guess is that this should be `requestService.mode === "website" && (appService.defaultLicensePoolId || appService.savedDefaultLicensePoolId)` so it's executed either if the field IS defined, or if it WAS defined.

File: source/javascripts/services/_appService.js.erb [hunk 3]
```diff
@@ -471,6 +496,29 @@
...
 			};
 
+			appService.getDefaultLicensePoolId = function() {
+				appService.defaultLicensePoolId =
+					appService.appConfig &&
+					appService.appConfig.meta &&
+					appService.appConfig.meta["<%= data[:constants][:meta_bundle_ids][:bitrise_io] %>"] &&
+					appService.appConfig.meta["<%= data[:constants][:meta_bundle_ids][:bitrise_io] %>"].license_pool_id;
+				if (!appService.defaultLicensePoolId) {
+					return $q.reject(new Error("<%= data[:strings][:machine_type][:invalid_set_in_bitrise_yml_meta] %>"));
```
Comment (erosdome) [inline source/javascripts/services/_appService.js.erb:506]: Is it indeed an error if a license pool ID is not set?

**Review by erosdome — CHANGES_REQUESTED** [review]

File: source/javascripts/controllers/_LicensesController.js.erb (hunk 1 above)
Comment (roland-spekker-bitrise) [reply source/javascripts/controllers/_LicensesController.js.erb:35]: I don't really understand why, but in some cases, it doesn't work without the manual $apply.  It is easy to reproduce the problem if you take out the manual $apply and try to refresh it a couple of times.

File: source/javascripts/controllers/_LicensesController.js.erb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+				},
+				function(error) {
+					$scope.licenses = [];
+					console.log(error);
```
[the remark this answers is not in the document]
Comment (roland-spekker-bitrise) [reply source/javascripts/controllers/_LicensesController.js.erb:39]: Good question, there was no error state in the figma for this. I'll ask, but this is for a demo for a prospect, I guess it's good for now. But I'll raise the issue in the project channel.

[the remark this answers is not in the document]
Comment (roland-spekker-bitrise) [reply source/javascripts/controllers/_LicensesController.js.erb:39]: Discussed it in the project channel and we'll add it in a later iteration.

File: source/javascripts/services/_appService.js.erb (hunk 2 above)
Comment (roland-spekker-bitrise) [reply source/javascripts/services/_appService.js.erb]: removed all default license pool related changes

File: source/javascripts/services/_appService.js.erb (hunk 3 above)
Comment (roland-spekker-bitrise) [reply source/javascripts/services/_appService.js.erb]: removed all default license pool related changes

**Review by erosdome — APPROVED** [review]

**Pull request merged**